openml/extensions/sklearn/extension.py

```diff
@@ -1609,16 +1606,31 @@ def _prediction_to_probabilities(y: np.ndarray, classes: List[Any]) -> np.ndarra
 
             Returns
             -------
-            np.ndarray
+            pd.DataFrame
             """
+
+            if isinstance(task, (OpenMLClassificationTask, OpenMLLearningCurveTask)):
```

Comment:
As far as I can tell this function is only called both of the first conditions are met (correct task type and has `class_labels`. It's a little hard to tell from Github though. But if that's correct, you can greatly simplify the flow of this function (and I would encourage you to do so).